### PR TITLE
Allow to configure file.encoding and user.language for native image building

### PR DESCRIPTION
**Pull request by gsmet**
Follow up of https://github.com/quarkusio/quarkus/pull/10527 .

/cc @jaikiran @blinkfox

### CODE REVIEW

File: core/deployment/src/main/java/io/quarkus/deployment/pkg/NativeConfig.java [hunk 1]
```diff
@@ -43,6 +45,36 @@
     @ConfigItem(defaultValue = "true")
     public boolean enableJni;
 
+    /**
+     * Defines the user language used for building the native executable.
+     * <p>
+     * Defaults to the system one.
+     */
+    @ConfigItem(defaultValue = "${user.language:}")
```
Comment (gsmet) [inline core/deployment/src/main/java/io/quarkus/deployment/pkg/NativeConfig.java:53]: @radcortez is it safe to have an empty default value with a non-Optional property? It seems to work but I'm a bit surprised :).

Comment (radcortez) [reply core/deployment/src/main/java/io/quarkus/deployment/pkg/NativeConfig.java:53]: That is definitely a bug :)

I guess we never covered that case. We probably just blindly accept the default value from the expansion without checking if there is something there. I'm going to add a fix for it, so I recommend to not lean on it :)

Comment (gsmet) [reply core/deployment/src/main/java/io/quarkus/deployment/pkg/NativeConfig.java:53]: Well, if I merge it before you fix it, it's your turn to fix it :).

I pushed a fix to use `Optional`.

Comment (radcortez) [reply core/deployment/src/main/java/io/quarkus/deployment/pkg/NativeConfig.java:53]: Ahahaha! Good job!

**Review by radcortez — APPROVED** [review]

**Comment by jaikiran** [comment]
Thanks @gsmet, this looks good to me.

**Comment by gsmet** [comment]
For now, I will only backport the first commit.

**Pull request merged**